server.js

```diff
@@ -1,33 +1,165 @@
-app.use(cors())
-app.use(bodyParser.json())
+import express from "express";
+import bodyParser from "body-parser";
+import cors from "cors";
+
+import spotifyData from "./data/spotify-releases.json";
+
+const port = process.env.PORT || 8080;
+const app = express();
+
+app.use(cors());
+app.use(bodyParser.json());
+
+// error message
+const ERROR_RELEASES_NOT_FOUND = { error: "No release matched your request" };
 
 // Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+app.get("/", (req, res) => {
+  res.send("Welcome to the Spotify Releases API");
+});
+
+// This route will return a collection of releases. It can be filtered by type, artist and title.
+app.get("/releases", (req, res) => {
+  const { type, artist, title } = req.query;
+
+  let filteredReleases = spotifyData;
+
+  if (type) {
+    filteredReleases = filteredReleases.filter(
+      (item) => item.album_type === type
+    );
+  };
+  if (artist) {
+    filteredReleases = filteredReleases.filter(
+      (item) => item.artists.map((artist) => artist.name).toString().includes(artist)
+    );
+  };
+  if (title) {
+    filteredReleases = filteredReleases.filter(
+      (item) => item.name && item.name.toString().includes(title)
+    );
+  };
+  if (filteredReleases.length === 0) {
+    res.status(404).json(ERROR_RELEASES_NOT_FOUND);
+  } else {
+    res.json({
+      total: filteredReleases.length,
+      releases: filteredReleases
+    });
+  }
+});
+
+// This route will return a single release based on id
+app.get("/releases/:id", (req, res) => {
+  const { id } = req.params;
+  const release = spotifyData.find(
+    (item) => item.id === id
+  );
+  if (!release) {
+    res.status(404).json(ERROR_RELEASES_NOT_FOUND);
+  } else {
+    res.json(release);
+  }
+});
+
+// This route will return a collection of releases for the specified artist
+app.get("/releases/artist/:artist", (req, res) => {
+  const { artist } = req.params;
+  const filteredReleases = spotifyData.filter(
+    (item) => item.artists.map((artist) => artist.name).toString() === artist
+  );
+  if (filteredReleases.length === 0) {
+    res.status(404).json(ERROR_RELEASES_NOT_FOUND);
+  } else {
+    res.json({
+      total: filteredReleases.length,
+      releases: filteredReleases
+    });
+  };
+});
+
+// This route will return a collection of releases in the specified market for the specified type
+app.get("/releases/market/:market/type/:type", (req, res) => {
+  const { market, type } = req.params;
+
+  let filteredReleases = spotifyData;
+
+  filteredReleases = filteredReleases.filter(
+    (item) => item.available_markets.includes(market)
+  );
+  filteredReleases = filteredReleases.filter(
+    (item) => item.album_type === type
+  );
+  if (filteredReleases.length === 0) {
+    res.status(404).json(ERROR_RELEASES_NOT_FOUND);
+  } else {
+    res.json({
+      total: filteredReleases.length,
+      releases: filteredReleases
+    });
+  };
+});
```

Comment:
These could also be queries under your first endpoint:
`/releases?type=album&market=SE`